server/src/main/java/org/elasticsearch/http/AbstractHttpServerTransport.java

```diff
@@ -349,6 +375,15 @@ private void handleIncomingRequest(final HttpRequest httpRequest, final HttpChan
             restRequest = innerRestRequest;
         }
 
+        final Logger trace;
+        if (tracerLog.isTraceEnabled() && shouldTraceRequest(httpRequest.uri())) {
+            tracerLog.trace(new ParameterizedMessage("[{}][{}][{}] received request from [{}]", restRequest.hashCode(),
```

Comment:
Instead of the hashCode here, I would prefer to generate an actual unique value for each request (AtomicLong generator could be in `RestRequest` AFAICS), and return that through a `getRequestId()` method.